web/src/components/SecureAdminConsole.jsx

```diff
@@ -214,23 +214,22 @@ class SecureAdminConsole extends React.Component {
     }
   };
 
-  componentDidMount() {
+  async componentDidMount() {
     window.addEventListener("keydown", this.submitForm);
 
-    const token = Utilities.getCookie("token");
-    if (token) {
+    const localStorageToken = window.localStorage.getItem("token");
+    const cookieToken = Utilities.getCookie("token");
+    const cookieSessionRole = Utilities.getCookie("session_role");
+    const localStorageSessionRole = window.localStorage.getItem("session_role");
+    if (Utilities.isLoggedIn() || cookieToken) {
       // this is a redirect from identity service login
       // strip quotes from token (golang adds them when the cookie value has spaces, commas, etc..)
```

Comment:
in the completeLogin function, we set token and session_role to localStorage. That's how we know if a user is logged in. 
However previously, on componentDidMount we retrieve token by cookies. (I believe that's why it would always prompt user to login) token is never set in cookies. Therefore, token was empty. 

However, I'm not comfortable removing cookies from this code, as the comment above mention it is a redirect from identity service login. Let me know what you think. :)